File: mobile/src/main/java/com/mattrubacky/monet2/api/splatnet/CoopResultRequest.java
package com.mattrubacky.monet2.api.splatnet;

import android.os.Bundle;

import com.mattrubacky.monet2.deserialized.splatoon.CoopResult;


import retrofit2.Response;

/**
 * Created by mattr on 10/24/2018.
 */

public class CoopResultRequest extends SplatnetRequest {
    private CoopResult job;
    private String id, referer;

    public CoopResultRequest(int id){
        this.id = String.valueOf(id);
        this.referer = "https://app.splatoon2.nintendo.net/coop_results/"+this.id;
    }

    @Override
    protected void manageResponse(Response response) {
        job = (CoopResult) response.body();
    }

    @Override
    public void setup(Splatnet splatnet, String cookie, String uniqueID) {
        call = splatnet.getSalmonResult(id,referer, cookie,uniqueID);
    }

    @Override
    public Bundle result(Bundle bundle){
        bundle.putParcelable("job",job);
        return bundle;
    }
}

File: mobile/src/main/java/com/mattrubacky/monet2/backend/api/splatnet/CoopResultRequest.kt
package com.mattrubacky.monet2.backend.api.splatnet

import com.mattrubacky.monet2.data.deserialized.splatoon.CoopResult
import retrofit2.Call

class CoopResultRequest(id:Int,splatnet:Splatnet,cookie:String,uniqueId:String): SplatnetRequest<CoopResult>(splatnet,cookie,uniqueId) {
    private val referer = "https://app.splatoon2.nintendo.net/coop_results/$id"
    override var call: Call<CoopResult> = splatnet.getSalmonResult(id.toString(),referer,cookie,uniqueId)
}